lisa/tools/firewall.py

```diff
@@ -60,27 +60,31 @@ def command(self) -> str:
     def can_install(self) -> bool:
         return False
 
-    def start_forwarding(self, port: int, dst_ip: str, dst_port: str) -> None:
-        self.run(
+    def start_forwarding(self, port: int, dst_ip: str, dst_port: int) -> None:
+        result = self.run(
             f"-I FORWARD -o virbr0 -p tcp -d {dst_ip} --dport {dst_port} -j ACCEPT",
-            sudo=True
+            sudo=True,
         )
+        result.assert_exit_code()
 
-        self.run(
+        result = self.run(
             f"-t nat -I PREROUTING -p tcp --dport {port} -j DNAT --to {dst_ip}:{dst_port}",  # noqa: E501
-            sudo=True
+            sudo=True,
         )
+        result.assert_exit_code()
 
-    def stop_forwarding(self, port: int, dst_ip: str, dst_port: str) -> None:
-        self.run(
+    def stop_forwarding(self, port: int, dst_ip: str, dst_port: int) -> None:
+        result = self.run(
             f"-D FORWARD -o virbr0 -p tcp -d {dst_ip} --dport {dst_port} -j ACCEPT",
-            sudo=True
+            sudo=True,
```

Comment:
Alternatively, you can add `expected_error_code=0` to the `run` call instead of using `assert_exit_code()`.